Formula/mlpack.rb

```diff
@@ -0,0 +1,93 @@
+class Mlpack < Formula
+  desc "Scalable C++ machine learning library"
+  homepage "https://www.mlpack.org"
+  url "https://mlpack.org/files/mlpack-3.2.2.tar.gz"
+  sha256 "7aef8c27645c9358262fec9ebba380720a086789d6519d5d1034346412a52ad6"
+
+  depends_on "cmake" => :build
+  depends_on "doxygen" => :build
+  depends_on "pkg-config" => :build
+  depends_on "armadillo"
+  depends_on "boost"
+  depends_on "ensmallen"
+  depends_on "graphviz"
+
+  def install
+    # stb
+    resource "stb" do
+      url "https://github.com/nothings/stb/archive/f67165c2bb2af3060ecae7d20d6f731173485ad0.tar.gz"
+      sha256 "ad5d34b385494cf68c52fad5762d00181c0c6d4787988fc75f17295c3c726bf8"
+    end
+    resource("stb").stage do
+      (include/"stb").install "stb.h"
+      (include/"stb").install "stb_easy_font.h"
+      (include/"stb").install "stb_leakcheck.h"
+      (include/"stb").install "stb_truetype.h"
+      (include/"stb").install "stb_c_lexer.h"
+      (include/"stb").install "stb_herringbone_wang_tile.h"
+      (include/"stb").install "stb_perlin.h"
+      (include/"stb").install "stb_voxel_render.h"
+      (include/"stb").install "stb_connected_components.h"
+      (include/"stb").install "stb_image.h"
+      (include/"stb").install "stb_rect_pack.h"
+      (include/"stb").install "stretchy_buffer.h"
+      (include/"stb").install "stb_divide.h"
+      (include/"stb").install "stb_image_resize.h"
+      (include/"stb").install "stb_sprintf.h"
+      (include/"stb").install "stb_ds.h"
+      (include/"stb").install "stb_image_write.h"
+      (include/"stb").install "stb_textedit.h"
+      (include/"stb").install "stb_dxt.h"
+      (include/"stb").install "stb_tilemap_editor.h"
+      (include/"stb").install "stb_vorbis.c"
+    end
+    # mlpack
+    cmake_args = std_cmake_args
+    cmake_args << "-DDEBUG=OFF"
+    cmake_args << "-DPROFILE=OFF"
+    cmake_args << "-DDOWNLOAD_STB_IMAGE=OFF"
+    cmake_args << "-DARMADILLO_INCLUDE_DIR=#{Formula["armadillo"].opt_include}"
+    cmake_args << "-DENSMALLEN_INCLUDE_DIR=#{Formula["ensmallen"].opt_include}"
+    cmake_args << "-DARMADILLO_LIBRARY=#{Formula["armadillo"].opt_lib}/libarmadillo.dylib"
+    cmake_args << "-DSTB_IMAGE_INCLUDE_DIR=#{(include/"stb")}"
+    cmake_args << "-DCMAKE_CXX_FLAGS=-fext-numeric-literals" unless ENV.compiler == :clang
```

Comment:
Why is `-fext-numeric-literals` needed? Doesn't `cmake` detect automatically what the compiler flags should be?